src/meetingsession/DefaultMeetingSession.ts

```diff
@@ -108,6 +109,15 @@ export default class DefaultMeetingSession implements MeetingSession {
     return this._deviceController;
   }
 
+  async destroy(): Promise<void> {
+    if (isDestroyable(this._deviceController)) {
+      await this._deviceController.destroy();
+    }
+    this._deviceController = undefined;
+    this.audioVideoController = undefined;
```

Comment:
Should we call `this.audioVideoController.stop()` here and then set to undefined?

Reply:
I actually feel like it should do `audioVideoController` (because the session created it) and _not_ the device controller, which was passed in as an argument.

What do you think?

Reply:
If you mean destroy the `audioVideoController` then yes, I think so it would be right to destroy what meeting session creates avoiding `deviceController` cleanup which is demo created. Further, the `cleanUpResources` function in the demo has some function calls which could be moved here like the `this.audioVideo.stopLocalVideoTile()` and so on which are not dependent on the objects created in the demo.

But I think, this cleanup at two places would be confusing. Also,  we have to explicitly document to call this at last as builders may call `meetingSession.destroy()` first and then this would set the `audioVideo` object to `undefined` and the demo cleanup calls will internally fail. So instead of `meetingSession.destroy()` should we just call `this.deviceController.destroy()` in the `cleanUpResources` itself in the demo?

Reply:
In writing this code I wanted it to be possible to use the SDK without explicitly destroying most of these core classes: it should be enough to remove observers and stop them as appropriate, and the demo serves to model that behavior. 

These `destroy` methods are a hammer to use when it is hard or impossible to untangle callbacks. Again the meeting demo showcases this when it explicitly destroys the POST loggers, which cannot otherwise be correctly disposed of.

Because the SDK classes typically promise to be reusable — that is, you can create a meeting, stop it, and then start it again — it's important that we clean up as we go, and do so as incrementally as possible. It should not be necessary for the demo to destroy its device controller in order to stop other classes from leaking.

Reply:
Okok got it! So, in-order to keep the re-usability here, will you be removing the `deviceController.destory()` call here or we need it to fix the leak?

Reply:
I've removed it and added destroys for the two owned controllers. This code is not called from the demo, so it's not necessary to fix leaks, because the demo cleans up on its own.